### PR TITLE
[XPU] Add xpu forward in Gemma3RMSNorm & Add test and benchmark for Gemma3RMSNorm

### PR DESCRIPTION
**Pull request by MeowMiaoJ**
## Motivation
This pull request adds comprehensive support and testing for the new `Gemma3RMSNorm` layer, since the 4d shape is accepted in sglang-kernel-xpu.

## Modifications

* Implemented the `forward_xpu` method in `Gemma3RMSNorm`, enabling XPU execution.
* Extended the test suite in `test_layernorm.py` with a new `TestGemma3RMSNorm` class, providing thorough correctness tests for the new layer across multiple shapes, data types, and residual addition scenarios, including non-contiguous ("unflattened") tensor cases.
* Added a microbenchmark script `bench_gemma3_rmsnorm.py` to measure and compare the performance of `Gemma3RMSNorm`'s native PyTorch and XPU-accelerated forward passes for various 2D/3D/4D tensor shapes, including non-contiguous layouts.

## Accuracy Tests

Two tests have passed to show the accuracy is not deducted.
`pytest -q /sgl-workspace/sglang/test/registered/kernels/test_fused_op_dispatch.py`

56 passed, 18 warnings in 20.21s

`pytest -q /sgl-workspace/sglang/test/manual/layers/test_layernorm.py`

5 passed, 3 skipped, 18 warnings, 194 subtests passed in 5.23s

## Speed Tests and Profiling

N/A

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :no_entry_sign: [Run #34335383885](https://github.com/sgl-project/sglang/actions/runs/34335383885)
Latest PR Test (Extra): :no_entry_sign: [Run #34455370107](https://github.com/sgl-project/sglang/actions/runs/34455370107)
Latest PR Test (AMD ROCm 10): :x: [Run #34335383792](https://github.com/sgl-project/sglang/actions/runs/34335383792)

### CODE REVIEW

**Review by CaoE — APPROVED** [review]
Please note the CI test results.

**Comment by CaoE** [comment]
Depends on https://github.com/sgl-project/sgl-kernel-xpu/pull/367.

**Comment by MeowMiaoJ** [comment]
> Depends on [sgl-project/sgl-kernel-xpu#367](https://github.com/sgl-project/sgl-kernel-xpu/pull/367).

Dependency PR is merged.

File: test/manual/layers/test_layernorm.py [hunk 1]
```diff
@@ -109,6 +114,174 @@ def test_gemma_rms_norm(self):
                 self._run_gemma_rms_norm_test(*params)
 
 
+class TestGemma3RMSNorm(CustomTestCase):
```
Comment (CaoE) [inline test/manual/layers/test_layernorm.py:117]: Can we reduce the size and choose only the smallest set to shorten the test time ?

File: test/manual/layers/test_layernorm.py [hunk 2]
```diff
@@ -3,7 +3,12 @@
 
 import torch
 
-from sglang.srt.layers.layernorm import GemmaRMSNorm, LayerNorm, RMSNorm
```
Comment (mingfeima) [inline test/manual/layers/test_layernorm.py:6]: this file is manual test, which will not be launched in CI.

Do you check locally? @MeowMiaoJ

Comment (MeowMiaoJ) [reply test/manual/layers/test_layernorm.py:6]: Yes. The pytest results before and after are aligned and there are no failed items.

**Review by mingfeima — APPROVED** [review]
